esmvalcore/preprocessor/_multimodel.py

```diff
@@ -176,13 +177,31 @@ def _datetime_to_int_days(cube):
         # wrong overlap indices
         # NOTE: this workaround is good only
         # for monthly data
-        real_date = datetime(date_obj.year, date_obj.month, 1, 0, 0, 0)
+        if overlap:
+            real_date = datetime(date_obj.year, date_obj.month, 1, 0, 0, 0)
+        else:
+            real_date = datetime(date_obj.year,
+                                 date_obj.month, 15, 0, 0, 0)
         real_dates.append(real_date)
 
+    # get the number of days starting from the reference unit
+    time_unit = cube.coord('time').units.name
+    time_offset = _get_time_offset(time_unit)
```

Comment:
Hey @valeriupredoi , this is one of the points that really confuses me. What happens if different cubes have different time offsets? Don't you want the offset with respect to a common reference time? I'm thinking something like 

```Python
def _datetime_to_int_days(cube, overlap=False):
    """Return list of int(days) with respect to a common reference.

    Cubes may have different calendars. This function extracts the date
    information from the cube and re-constructs a default calendar using
    the datetime library.

    This doesn't work for daily or subdaily data, because different
    calendars may have different number of days in the year.

    The standard calendar used by datetime provides the to_ordinal()
    function which returns "int number of days since the 0001/01/01".
    """
    # Convert to a common time format
    years = [cell.point.year for cell in cube.coord('time').cells()]
    months = [cell.point.month for cell in cube.coord('time').cells()]
    if not 0 in np.diff(years):
        # yearly data
        standard_dates = [datetime(year, 7, 1) for year in years]
    elif not 0 in np.diff(months):
        # monthly data
        standard_dates = [datetime(year, month, 15) for year, month in zip(years, months)]
    else:
        raise ValueError("Multimodel only supports yearly or monthly data")

    return [dt.to_ordinal() for dt in standard_dates]
```

I can work this out further if you like?

Reply:
yes that's exactly that, I use the number of days since the start of the data taken from units forcing a `STANDARD_CALENDAR` eg `days since 1950-01-01, calendar=STANDARD_CALENDAR` - this works fine if there is no variation in calendars since the units are all the same for all cubes after `cmor.fix` unified the units, but I suspect it's the cause of the shifts @LisaBock is seeing when calendars differ

Reply:
btw isn't it `toordinal()`?

Reply:
> btw isn't it `toordinal()`?

Right you are :beer:

Reply:
> this works fine if there is no variation in calendars

So I'd rather just fix it, either to the ordinal or to 1950/01/01. Taking it from the cube seems unnecessary and error-prone.

Reply:
I fail to see the error-prone nature if all the cubes have the same units as they should have. Also taking it from the cube assumes nothing, it's safer in my books; I believe the stuff that Lisa is seeing has to do with cmip3 data not being correctly fixed

Reply:
> I fail to see the error-prone nature if all the cubes ...

It's in the "if", right?

> taking it from the cube assumes nothing

Yes, it assumes the calendars offsets have been properly fixed.

> it's safer in my books

I don't agree. You don't need the new time array to have anything to do with the old cubes, except for it to represent the same dates (which you get using the `cell.point.year` and `cell.point.month` methods . So why not just convert all of them to the same, easy to work with calendar?

Reply:
ah right, I only saw now that you are constructing the full datetime object - yes that should work, cheers! Can you pls commit and run the tests? It should pass them

Reply:
Don't cheer too early, this doesn't pass the tests yet. It removes the overlap argument again, and also the `_get_time_offset` function. Then, the tests assume a different ordinal for the `expected` cubes, and also I think it returns a list of integers, where the old version returned `datetime.timedelta` objects. But I guess I can fix all this and make a commit, if you like.

Reply:
the thing is - we shouldn't need this (apart from the nicer check on determining if it's yearly data, yes) - it all works fine as it is, unless there are issues with the actual ingested data in the MM as I'm suspecting it's Lisa's case. She is running CMIP3, we never had any MM issues with the MM for CMIP5/6

Reply:
Maybe we don't... but you have to agree that this thing has been a pain to debug, right? So the cleaner we make it, the easier it should get. Let me invest a bit more time, and see if it solves Lisa's problem. I don't have other V2 deadlines ;-)

Reply:
yeah not the best code to understand indeed! but it does do a lot of tricksy stuff nonetheless. Can I suggest we finish up with this PR in terms of functionality for the yearly (normal) data and merge then we look into Lisa's issue, the more I look at it the more it looks to be way separate :beer:

Reply:
I agree we should respect the scope of the PR, but let's not merge to hastily. I'm not so keen on the overlap argument for the yearly data. But removing it in favor of the above suggestion breaks several tests. Can I look into fixing that first?

Reply:
that would ve very much appreciated mate and will buy a :beer:

Reply:
So I fixed most of it, but some tests still fail because test cube 2 has a *daily* time coordinate, and the code above raises an exception in that case. What to do with that?

Reply:
let's not make the changes - your suggestion below saved the day, so let's keep the changes to a minimum and only commit what's necessary - let me take it from here :beer: